CleverCloud/clever-components
# fix(cc-ssh-key-list): proper form validation when adding a new key in some cases

Fixes #632, Fixes #633.

### What this PR fixes

In the component creation form, model and view are getting out of sync in some cases (check the very detailed issues for more details).

### How to review

Check the code : the fix is quite straightforward (alongside some minor refactor).

UI test 1 :
- in the creation form, enter a random key name ("The front-end team is awesome" for example" :fire:)
- enter white-spaces `    `  in the public key field
- what was happening (on submit) : the white-spaces are still visible in the input, the error message for an empty input is shown
- what is now expected (on submit) : the input is cleared, the error message for an empty input is still shown

UI test 2 :
- in the creation form, enter a random key name ("Vote for Bob" for example" :eyes:)
- enter the following string `    this is a private key    ` (with the trailing white-spaces)  in the public key field
- what was happening :
	- (on first submit) the input shows the string *without* the trailing white-spaces, the error message for a private key is shown
	- at this point, clear and enter the initial string again (with the trailing white-spaces) in the public key field
	- (on second submit) the input shows the string *with* the trailing white-spaces, the error message for a private key is still shown
- what is now expected (on each submit) : the input now shows the trimmed string (`this is a private key`), the error message for a private key is still shown

PS : don't forget you can use the `demo-smart/index.html` page.
